### PR TITLE
OD-539 [FIX] Fix - settings aren't loaded

### PR DESCRIPTION
**Pull request by OleksandrLavrynenko**
@sofiiakvasnevska @inna-bieshulia

## Issue
OD-539 https://weboo.atlassian.net/browse/OD-539

## Description
The component loading is fixed when DataSource is in trash

## Screenshots/screencasts
https://storyxpress.co/video/kqw26g55oni5zutas

## Backward compatibility
This change is fully backward compatible.

## Reviewers
@upplabs-alex-levchenko

### CODE REVIEW

File: js/interface.js [hunk 1]
```diff
@@ -937,16 +941,26 @@ var DynamicLists = (function() {
...
+
+            return;
+          }).catch(function() {
+            _this.initDefaultDatasourceProvider();
+
             resolve();
-          });
 
-          initDataSourceProvider(_this.config.dataSourceId);
+            return;
```
Comment (tonytlwu) [inline js/interface.js:962]: Why is this `return` necessary?

File: js/interface.js [hunk 2]
```diff
@@ -937,16 +941,26 @@ var DynamicLists = (function() {
...
-          $document.on('datasource-selected', function() {
+          // Check data source
+          _this.getDataSourceById(_this.config.dataSourceId).then(function() {
+            $document.on('datasource-selected', function() {
+              resolve();
+            });
+
+            _this.initDataSourceProvider(_this.config.dataSourceId);
+
+            return;
```
Comment (tonytlwu) [inline js/interface.js:956]: Why is this `return` necessary?

**Review by tonytlwu — CHANGES_REQUESTED** [review]
Bad PR title

Rejected +1

File: js/interface.js (hunk 1 above)
Comment (OleksandrLavrynenko) [reply js/interface.js:962]: > Why is this `return` necessary?

Apologies, this code was from previous implementation. I'll fix that

File: js/interface.js (hunk 2 above)
Comment (OleksandrLavrynenko) [reply js/interface.js:956]: > Why is this `return` necessary?

Apologies, this code was from previous implementation. I'll fix that

File: js/interface.js [hunk 3]
```diff
@@ -894,6 +894,10 @@ var DynamicLists = (function() {
           _this.initializeDetailViewSortable();
         });
     },
+    initDefaultDatasourceProvider () {
```
Comment (squallstar) [inline js/interface.js:897]: This is ES6 syntax. We need to support IE11.

**Review by squallstar — CHANGES_REQUESTED** [review]
Rejected +1

File: js/interface.js (hunk 3 above)
Comment (OleksandrLavrynenko) [reply js/interface.js]: > This is ES6 syntax. We need to support IE11.

Already fixed

File: js/interface.js [hunk 4]
```diff
@@ -937,16 +941,22 @@ var DynamicLists = (function() {
...
           resolve();
         } else {
-          $document.on('datasource-selected', function() {
+          // Check data source
+          _this.getDataSourceById(_this.config.dataSourceId).then(function() {
+            $document.on('datasource-selected', function() {
+              resolve();
+            });
+
+            _this.initDataSourceProvider(_this.config.dataSourceId);
```
Comment (squallstar) [inline js/interface.js:954]: It doesn't look like this function is defined in the `_this` scope. Did you test your code?

Rejected +1

**Review by squallstar — CHANGES_REQUESTED** [review]

File: js/interface.js (hunk 4 above)
Comment (OleksandrLavrynenko) [reply js/interface.js:954]: > It doesn't look like this function is defined in the `_this` scope. Did you test your code?
> 
> Rejected +1

Code is tested and completely work

Comment (OleksandrLavrynenko) [reply js/interface.js]: > It doesn't look like this function is defined in the `_this` scope. Did you test your code?
> 
> Rejected +1

I'll fix that

**Review by squallstar — APPROVED** [review]
@sofiiakvasnevska this is ready to be deployed on **staging** for QA.

- **Risk**: Medium
- **Affected**: Studio users configuring the LFD
- **Verify**: Verify the issue is resolved

**Pull request merged**